assets/javascripts/modules/modalDialog.js

```diff
@@ -0,0 +1,351 @@
+   * @param {Function} setMaskDimensions - callback function
+   */
+  $(window).on('resize', function() {
+    setModalDialogDimensions();
+    setDimensions($(contentShowing));
+  });
+
+  /**
+   * Event handler to capture [esc] key press in modal-dialog
+   **/
+  $modalDialog.on('keyup', contentShowing, function(e) {
+    // escape key press to close modal-dialog
+    if (isModalDialogOpen() && e.which === 27) {
+      // send google analytics event - assume the same value as the close button
+      var $closeButton = $(contentShowing).find('[data-modaldialog-action="close"]');
+
+      if ($closeButton.length)
+      {
+        fireEventTracking($closeButton.data('journeyClick'));
+      }
+
+      hide();
+    }
+  });
+
+  /**
+   * Event handler to capture tab key press in modal-dialog - needs to be in doc context to capture tab outside modal-dialog.
+   */
+  $doc.on('keydown', function(e) {
+    if (isModalDialogOpen() && e.which === 9) {
+      // listen for tab direction event here
+      e.preventDefault();
+      var $src = $(e.target),
+        isBack = e.shiftKey,
+        $tabs = $(contentShowing).find('[tabindex="1"]').addBack(),
+        inc = $tabs.index($src) + (isBack ? -1 : 1),
+        $focus;
+
+      // if focus has moved out of modal-dialog bring it back
+      if (inc < 0) {
+        $focus = $tabs.last();
+      }
+      else if (inc > $tabs.length - 1) {
+        $focus = $tabs.first();
+        $container.scrollTop(0);
+      }
+      else {
+        $focus = $tabs.eq(inc);
+      }
+
+      $focus.focus();
+    }
+  });
+
+  /**
+   * Event handler for document click to return focus to modal-dialog when content showing.
+   */
+  $doc.on('click', function(e) {
+    if (isModalDialogOpen() && !$(contentShowing).find($(e.target)).length) {
+      if ($(contentShowing).find('[tabindex="1"]:focus').length) {
+        $(contentShowing).find('[tabindex="1"]:focus').focus();
+      } else {
+        $(contentShowing).focus();
+        e.stopPropagation();
+      }
+    }
+  });
+
+  /**
+   * Event handler for hiding modal-dialog & content.
+   */
+  $body.find($contents).addBack().on('click', '[data-modaldialog-action="close"]', function(e) {
+    e.preventDefault();
+    hide();
+  });
+
+  /**
+   * Event handler for showing modal-dialog & content.
+   */
+  $body.find($contents).addBack().on('click', '[data-modaldialog-action="open"]', function(e) {
+    e.preventDefault();
+    setModalDialogDimensions();
+    show(e);
+  });
+
+  /**
+   * Event handler for custom event triggered by hidden attribute / prop state change.
+   */
+  $contents.on('hiddenChange', function(e, isHidden) {
+    if (isHidden) {
+      if (!$(contentShowing).length) {
+        hide();
+      }
+    } else {
+      show();
+    }
+  });
+
+  /**
+   * Setup elements
+   */
+  function setup() {
+    $contents.each(function() {
+      var $this = $(this);
+      $this.setHidden($this.getHidden());
+    });
```

Comment:
Maybe a preference here but can you comment through code? as in variable and function names, people tend not to read comments right?